powerauth-webflow-authentication-sms/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/sms/controller/SMSAuthorizationController.java

```diff
@@ -71,7 +71,10 @@ protected String authenticate(SMSAuthorizationRequest request) throws AuthStepEx
         final GetOperationDetailResponse operation = getOperation();
         Logger.getLogger(this.getClass().getName()).log(Level.INFO, "Step authentication started, operation ID: {0}, authentication method: {1}", new String[] {operation.getOperationId(), getAuthMethodName().toString()});
         checkOperationExpiration(operation);
-        final Object messageId = httpSession.getAttribute(MESSAGE_ID);
+        final Object messageId;
+        synchronized (httpSession.getServletContext()) {
```

Comment:
Can this be written in some better way? I am not sure if it is a good idea to use `synchronized` keyword in the server-side code... if there is a better way, let's improve it in all instances...